### PR TITLE
doc: Add initial 'Cryptography' section

### PR DESCRIPTION
**Pull request by slyon**
## Description
Add initial 'Cryptography' section do our "Security" documentation.

Also include some new tooling in `tools/grep_cryptography.sh`, to identify cryptography related terminology in our codebase.

FR-7556

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains code coverage (`make check-coverage`).
- [x] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

### CODE REVIEW

**Comment by slyon** [comment]
/review
--pr_reviewer.inline_code_comments=true
--pr_reviewer.require_score_review=true
--pr_reviewer.require_can_be_split_review=true
--pr_reviewer.require_soc2_ticket=true
--pr_reviewer.num_code_suggestions="2"

**Comment by slyon** [comment]
/review
--pr_reviewer.inline_code_comments=true
--pr_reviewer.require_score_review=true
--pr_reviewer.require_can_be_split_review=true
--pr_reviewer.require_soc2_ticket=true
--pr_reviewer.num_code_suggestions="2"

**Comment by slyon** [comment]
/review
--pr_reviewer.inline_code_comments=true
--pr_reviewer.require_score_review=true
--pr_reviewer.require_can_be_split_review=true
--pr_reviewer.require_soc2_ticket=true
--pr_reviewer.num_code_suggestions="2"

File: src/validation.c [hunk 1]
```diff
@@ -71,7 +71,7 @@ is_hostname(const char *hostname)
 gboolean
 is_wireguard_key(const char* key)
 {
-    /* Check if this is (most likely) a 265bit, base64 encoded wireguard key */
+    /* Check if this is (most likely) a 256 bit, base64 encoded wireguard key */
```
[the remark this answers is not in the document]
Comment (slyon) [reply src/validation.c:74]: unrelated to this PR

File: doc/security.md [hunk 2]
```diff
@@ -36,6 +36,61 @@ will become world-readable.
 * `/run/systemd/system/netplan-wpa-*.service`
 * `/run/systemd/system/systemd-networkd-wait-online.service.d/10-netplan*.conf`
 
+## Cryptography
+
+Netplan does not directly utilise cryptography, but configures underlying tools
+to do so. Such tools include `wpa_supplicant`, `systemd-networkd`, `NetworkManager`
+or `Open vSwitch` and they can for example be configured to setup WPA2/WPA3
+[encypted WiFi](https://w1.fi/wpa_supplicant/devel/code_structure.html#crypto_func) 
+connections, `802.1x` encrypted Ethernet connections,
```
Comment (daniloegea) [inline doc/security.md:45]: We probably should reword this as something like `802.1x for wired and wireless authentication and authorization,` as 802.1x alone doesn't provide encryption of network traffic.

**Review by daniloegea — COMMENTED** [review]
I left 2 comments in the text that we might want to address

**Review by daniloegea — APPROVED** [review]
Thanks, it looks good to me.
